crypto/evp/e_chacha20_poly1305.c

```diff
@@ -347,9 +350,10 @@ static int chacha20_poly1305_ctrl(EVP_CIPHER_CTX *ctx, int type, int arg,
         if (actx) {
             EVP_CIPHER_CTX *dst = (EVP_CIPHER_CTX *)ptr;
 
-            dst->cipher_data =
-                   OPENSSL_memdup(actx, sizeof(*actx) + Poly1305_ctx_size());
-            if (dst->cipher_data == NULL) {
+            EVP_CIPHER_CTX_set_cipher_data(dst,
+                    OPENSSL_memdup(actx, POLY1305_actx_size(actx)));
+
+            if (EVP_CIPHER_CTX_get_cipher_data(dst) == NULL) {
```

Comment:
Because you will end up with ugly indentation, I suggest this instead:

``` C
            actx = OPENSSL_memdup(actx, sizeof(*actx) + Poly1305_ctx_size());
            EVP_CIPHER_CTX_set_cipher_data(dst, actx);

            if (actx == NULL) {
```

(either way, you changed the size to `OPENSSL_memdup`)

Reply:
Addressed.